scripts/linter/collect-docs.js

```diff
@@ -0,0 +1,32 @@
+const fsPromises = require("fs").promises;
+const path = require("path");
+
+/**
+ * Walks the content directory tree (`start`) yielding Markdown files.
+ */
+async function* walkContent(start = "content") {
+    const files = await fsPromises.readdir(start, { withFileTypes: true });
+
+    for (const f of files) {
+        if (f.name.endsWith(".md")) {
+            yield path.join(start, f.name);
+        } else if (f.isDirectory()) {
+            yield* walkContent(path.join(start, f.name));
+        }
+    }
+}
+
+/**
+ * Return an array of paths to Markdown files.
+ */
+async function collectDocs() {
+    const candidateDocs = [];
+
+    for await (const candidateDoc of walkContent()) {
+        candidateDocs.push(candidateDoc);
+    }
```

Comment:
Interesting results for globbing. Since, as you say, it doesn't matter, I won't rewrite it to be synchronous, but I will keep that in mind for the future.

> It's not wrong but I don't understand this JS. The `walkContent` is asynchronous, right? So what was the point of then turning it into a big array?

`collectDocs` used to do some other things beside squashing `walkContent`, but when we dropped `meta.yaml` files, building an array was all that was left. I've unrolled that bit. Now, `walkDocs` yields each Markdown file asynchronously (had to rename the module for clarity and consistency though).